### PR TITLE
cpu/stm32_common/i2c: Fix error handling in i2c_1.c

### LINKED ISSUE #10254: stm_common/i2c1.c: I2C cannot recover from error for
#### Description
STM32 F0, F3, F7, L0 and L4 families will be locked after an error is set (nack for example).

One problem is in the _start, `while (!(i2c->CR2 & I2C_CR2_START) && tick--) {}` is waiting for the bit to be high, but it should wait for the bit to go low.

In the _stop, the stop bit never gets sent if there is an error, it should return after the error.

Also the error flags don't get cleared.

#### Steps to reproduce the issue
1. read reg without wires connected
2. read reg with wires connected
#### Expected results
1. ENXIO
2. valid data

#### Actual results
1. EIO
2. ETIMEDOUT

### PR DESCRIPTION
**Pull request by MrKevinWeiss**
### Contribution description

Errors flags could not clear making the i2c unusable after error.
This fix removes the error check in start so the error flags can clear and does proper checking for status bits before _bus_check.

### Testing procedure
Run periph_i2c on any STM32 F0, F3, F7, L0 and L4 family.  I connect a PHiLIP but any i2c sensor can be used.
`BOARD=nucleo-l476rg make flash term -C tests/periph_i2c`
```
i2c_read_regs 0 85 152 10 0    
2018-10-26 09:45:14,539 - INFO # i2c_read_regs 0 85 152 10 0
2018-10-26 09:45:14,545 - INFO # Command: i2c_read_regs(0, 0x55, 0x98, 10, 0x00)
2018-10-26 09:45:14,557 - INFO # Success: i2c_0 read 10 byte(s) from reg 0x98 : [0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08, 0x09]

i2c_read_regs 0 84 152 10 0    
2018-10-26 09:46:33,204 - INFO #  i2c_read_regs 0 84 152 10 0
2018-10-26 09:46:33,206 - INFO # Command: i2c_read_regs(0, 0x54, 0x98, 10, 0x00)
2018-10-26 09:46:33,216 - INFO # Error: ENXIO [6]

i2c_read_regs 0 85 152 10 0    
2018-10-26 09:46:49,596 - INFO #  i2c_read_regs 0 85 152 10 0
2018-10-26 09:46:49,601 - INFO # Command: i2c_read_regs(0, 0x55, 0x98, 10, 0x00)
2018-10-26 09:46:49,620 - INFO # Success: i2c_0 read 10 byte(s) from reg 0x98 : [0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08, 0x09]

i2c_write_regs 0 85 152 0 42   
2018-10-26 09:47:33,078 - INFO #  i2c_write_regs 0 85 152 0 42
2018-10-26 09:47:33,083 - INFO # Command: i2c_write_regs(0, 0x55, 0x98, 0x00, [0x2a])
2018-10-26 09:47:33,085 - INFO # Success: i2c_0 wrote 1 bytes to reg 0x98
```

### Issues/PRs references

Fixes #10254

### CODE REVIEW

File: cpu/stm32_common/periph/i2c_1.c [hunk 1]
```diff
@@ -475,14 +470,14 @@ static inline int _stop(I2C_TypeDef *i2c)
     if (i2c->ISR & ERROR_FLAG || !tick) {
         return -EIO;
     }
-
+
```
Comment (aabadie) [inline cpu/stm32_common/periph/i2c_1.c]: trailing whitespace ? ;)

**Comment by MrKevinWeiss** [comment]
@leandrolanzieri can test here if needed.

**Comment by MrKevinWeiss** [comment]
@aabadie my only issue is removing the timeout on the _check_bus may have some unexpected consequences.  I think I wait on the right flags though and it seems to read/write OK.

File: cpu/stm32_common/periph/i2c_1.c (hunk 1 above)
Comment (MrKevinWeiss) [reply cpu/stm32_common/periph/i2c_1.c]: shhhh, I think I force pushed pretty fast :)

**Comment by jia200x** [comment]
Thanks for this fix! We can merge then

**Pull request merged**